drivers/l3gd20/l3gd20.py

```diff
@@ -0,0 +1,209 @@
+L3GD20_BLE_LSB                       = const(0x00) 
+L3GD20_BLE_MSB                       = const(0x40) 
+L3GD20_HIGHPASSFILTER_DISABLE        = const(0x00) 
+L3GD20_HIGHPASSFILTER_ENABLE         = const(0x10) 
+L3GD20_INT1                          = const(0x00) 
+L3GD20_INT2                          = const(0x01) 
+L3GD20_INT1INTERRUPT_DISABLE         = const(0x00) 
+L3GD20_INT1INTERRUPT_ENABLE          = const(0x80) 
+L3GD20_INT2INTERRUPT_DISABLE         = const(0x00) 
+L3GD20_INT2INTERRUPT_ENABLE          = const(0x08) 
+L3GD20_INT1INTERRUPT_LOW_EDGE        = const(0x20) 
+L3GD20_INT1INTERRUPT_HIGH_EDGE       = const(0x00) 
+L3GD20_BOOT_NORMALMODE               = const(0x00) 
+L3GD20_BOOT_REBOOTMEMORY             = const(0x80) 
+L3GD20_HPM_NORMAL_MODE_RES           = const(0x00) 
+L3GD20_HPM_REF_SIGNAL                = const(0x10) 
+L3GD20_HPM_NORMAL_MODE               = const(0x20) 
+L3GD20_HPM_AUTORESET_INT             = const(0x30) 
+L3GD20_HPFCF_0                       = const(0x00)
+L3GD20_HPFCF_1                       = const(0x01)
+L3GD20_HPFCF_2                       = const(0x02)
+L3GD20_HPFCF_3                       = const(0x03)
+L3GD20_HPFCF_4                       = const(0x04)
+L3GD20_HPFCF_5                       = const(0x05)
+L3GD20_HPFCF_6                       = const(0x06)
+L3GD20_HPFCF_7                       = const(0x07)
+L3GD20_HPFCF_8                       = const(0x08)
+L3GD20_HPFCF_9                       = const(0x09)
+#
+# Default configuration:
+# Output data rate  190 Hz
+# Bandwidth/Cut off  50 Hz
+L3GD20_CTRL_REG1_VAL = const(0b01101111)
+# Normal Mode
+# Highpass cut-off 0.018Hz
+L3GD20_CTRL_REG2_VAL = const(0b00001001)
+# No interrupts
+L3GD20_CTRL_REG3_VAL = const(0b00000000)
+# Continous block update
+# Little endinan data
+# Full scale is 250 dps (degrees per second)
+# Use 4 wire SPI
+L3GD20_CTRL_REG4_VAL = const(0b00000000)
+# Normal Mode
+# FIFO is enabled
+# No Highpass
+# Int1/int2:00
+#
+L3GD20_CTRL_REG5_VAL = const(0b01000000)
+
+
+class L3GD20:
+    #
+    # Debug 
+    #
+    DEFAULT_CONF = [ 
+    (L3GD20_CTRL_REG1_ADDR, L3GD20_CTRL_REG1_VAL),
+    (L3GD20_CTRL_REG2_ADDR, L3GD20_CTRL_REG2_VAL),
+    (L3GD20_CTRL_REG3_ADDR, L3GD20_CTRL_REG3_VAL),
+    (L3GD20_CTRL_REG4_ADDR, L3GD20_CTRL_REG4_VAL),
+    (L3GD20_CTRL_REG5_ADDR, L3GD20_CTRL_REG5_VAL)]
+    DEBUG = False
+    def __init__(self, cs='PC1', spiNr=5):
+        self._conf={}
+        self.cs_pin = Pin(cs, Pin.OUT_PP, Pin.PULL_NONE)
+        self.cs_pin.high()
+        self.spi = SPI(spiNr, SPI.MASTER, baudrate=328125, polarity=1, phase=1, bits=8)
+        self.who_am_i = self.read_id()
+        if self.who_am_i == L3GD20_I_AM_L3GD20:
+            for addr, val in L3GD20.DEFAULT_CONF:
+                self.write_bytes(addr, bytearray([val,]))
+                self._conf[addr] = val
+        else:
+            raise Exception('L3GD20 gyro not present')
+        self.updateDpsFS()
+
+    def updateDpsFS(self):
```

Comment:
If this is only ever called from init then it should be embedded in init.  Otherwise it should be called `update_dps_fs`.